src/lib/operators/get_table.cpp

```diff
@@ -240,7 +240,7 @@ std::shared_ptr<const Table> GetTable::_on_execute() {
   }
 
   return std::make_shared<Table>(pruned_column_definitions, TableType::Data, std::move(output_chunks),
-                                 stored_table->uses_mvcc());
+                                 stored_table->uses_mvcc(), stored_table->get_table_indexes());
```

Comment:
We could add a parameter to the GetTable constructor (e.g., `include_index` with `false` as the default value). Similarly, we could add a corresponding member variable in the StoredTableNode: if this value is `true` in the StoredTableNode, we also set the corresponding value in the GetTable operator to `true` when the StoredTableNode is translated to the GetTable operator. Only then will the indexes be added to the returned table. Ideally, the optimizer should decide when a JoinNode (LQP) is translated into a JoinIndex (PQP). We currently don't have this rule. If such a rule decides to use an IndexJoin, then this rule should also set the flag in the corresponding StoredTableNode.

Pinging @Bouncner @tjjordan

Reply:
Hmm, I am not a big fan of default values. But first of all, I haven't thought about this issue enough.

The issue is, that we need to "link" an existing secondary to a table. Since the GetTable operator create a new temporary table, we also need to link to the index. So far correct?

What bothers me is that we don't check if the index is actually applicable here. If we introduce table indexes, why pass them if the corresponding column is no longer part of the table or the indexed chunks are pruned?

I am still not sure about the interface of `Table`. But maybe let's start first with the issue of "qualifying indexes"?

Reply:
How do you like the solution from commit 85dfceffecb52e5c8d7f979c58288c4cd6674119?